contrib/node/src/python/pants/contrib/node/subsystems/mixins.py

```diff
@@ -0,0 +1,64 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.contrib.node.subsystems.command import command_gen
+
+
+PACKAGE_MANAGER_NPM = 'npm'
+PACKAGE_MANAGER_YARNPKG = 'yarnpkg'
+
+
+class PackageManagerMixin(object):
+  """Defines node package manager functionalities."""
+
+  def _get_installation_args(self, install_optional):
+    """Returns command line args for installing package.
+
+    :param install_optional: True to request install optional dependencies.
+    :rtype: list of strings
+    """
+    raise NotImplementedError
+
+  def _get_run_script_args(self):
+    """Returns command line args to run a package.json script.
+
+    :rtype: list of strings
+    """
+    raise NotImplementedError
+
+  def install_packages(self, install_optional=False, node_paths=None):
```

Comment:
Consider adding an args parameter as well. Perhaps we only want to install with --only=production, --force, --dry-run as a few examples.

Reply:
added: --force and --production.